### DISCUSSION TITLE
Should the function have the NA_character of NA_real?

### DISCUSSION THREAD

**Discussion by lanawwas** [impact-initiatives-hppu/humind#410] [General]
@gnoblet In these lines of the add_comp_prot.R code 

https://github.com/impact-initiatives-hppu/humind/blob/02a748112bd64760e1ae69d4bbfd7df565e44a5a/R/add_comp_prot.R#L46C1-L47C33

As in the current code, mutate will mix between character and real values, wouldn't be better to have the NA_real instead of NA_character. What do you think? I keep it here as a discussion while the unit-test will throw an error for now for the tests with the following message 

```r
==> Testing R file using 'testthat'

ℹ Loading humind

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 0 ]
[ FAIL 1 | WARN 0 | SKIP 0 | PASS 0 ]
[ FAIL 2 | WARN 0 | SKIP 0 | PASS 0 ]
[ FAIL 3 | WARN 0 | SKIP 0 | PASS 0 ]
[ FAIL 3 | WARN 0 | SKIP 0 | PASS 1 ]
[ FAIL 4 | WARN 0 | SKIP 0 | PASS 1 ]
[ FAIL 5 | WARN 0 | SKIP 0 | PASS 1 ]
[ FAIL 5 | WARN 0 | SKIP 0 | PASS 2 ]
[ FAIL 6 | WARN 0 | SKIP 0 | PASS 2 ]
[ FAIL 7 | WARN 0 | SKIP 0 | PASS 2 ]

── Error ('test-add_comp_prot.R:15:3'): add_comp_prot function works correctly with default parameters ──
<dplyr:::mutate_error/rlang_error/error/condition>
Error in `dplyr::mutate(df, comp_prot_child_sep_cat = dplyr::case_when(!!rlang::sym(child_sep_cat) %in% 
    "none" ~ 1, !!rlang::sym(child_sep_cat) %in% "very_severe" ~ 
    5, !!rlang::sym(child_sep_cat) %in% "severe" ~ 4, !!rlang::sym(child_sep_cat) %in% 
    "non_severe" ~ 1, !!rlang::sym(child_sep_cat) %in% "undefined" ~ 
    NA_character_, .default = NA_character_))`: ℹ In argument: `comp_prot_child_sep_cat = dplyr::case_when(...)`.
Caused by error in `dplyr::case_when()`:
! Can't combine `..1 (right)` <double> and `..5 (right)` <character>.

```

Propose changes are : 

- Changed` NA_character_ to NA_real_ to ensure consistent numeric type.

- Changed .default = NA_character_ to .default = NA_real_ in both case_when statements.

**Comment by gnoblet** [comment]
Yes, `NA_real_` for these as eventually we need to calculate average of scores for instance. Discrepancy.

**Comment by lanawwas** [comment]
Solved in 59efc98
